sentinelhub/sentinelhub_batch.py

```diff
@@ -0,0 +1,398 @@
+        :type user_id: str or None
+        :param config: A configuration object
+        :type config: SHConfig or None
+        :param kwargs: Any additional parameters to include in a request query
+        :return: An iterator over existing batch requests
+        :rtype: Iterator[SentinelHubBatch]
+        """
+        url = SentinelHubBatch._get_process_url(config)
+        params = _remove_undefined_params({
+            'search': search,
+            'sort': sort,
+            'userid': user_id,
+            **kwargs
+        })
+        for request_info in _iter_pages(url, **params):
+            yield SentinelHubBatch(request_info=request_info, config=config)
+
+    @staticmethod
+    def get_latest_request(config=None):
+        """ Provides a batch request that has been created the latest
+        """
+        # This should be improved once sort parameter will be supported
+        batch_requests = list(SentinelHubBatch.iter_requests(config=config))
+        return max(*batch_requests, key=lambda request: request.info['created'])
+
+    def delete(self):
+        """ Delete a batch job request
+        """
+        url = self._get_process_url(self.config, request_id=self.request_id)
+        return get_auth_json(url, request_type=RequestType.DELETE)
+
+    def start_analysis(self):
+        """ Starts analysis of a batch job request
+        """
+        return self._call_job('analyse')
+
+    def start_job(self):
+        """ Starts running a batch job
+        """
+        return self._call_job('start')
+
+    def cancel_job(self):
+        """ Cancels a batch job
+        """
+        return self._call_job('cancel')
+
+    def restart_job(self):
+        """ Restarts only those parts of a job that failed
+        """
+        return self._call_job('restartpartial')
+
+    def iter_tiles(self, status=None, **kwargs):
+        """ Iterate over info about batch request tiles
+
+        :param status: A filter to obtain only tiles with a certain status
+        :type status: str or None
+        :param kwargs: Any additional parameters to include in a request query
+        :return: An iterator over information about each tile
+        :rtype: Iterator[dict]
+        """
+        url = self._get_tiles_url()
+        params = _remove_undefined_params({
+            'status': status,
+            **kwargs
+        })
+        return _iter_pages(url, **params)
+
+    def get_tile(self, tile_id):
+        """ Provides information about a single batch request tile
+
+        :param tile_id: An ID of a tile
+        :type tile_id: int or None
+        :return: Information about a tile
+        :rtype: dict
+        """
+        url = self._get_tiles_url(tile_id=tile_id)
+        return get_auth_json(url)
+
+    def reprocess_tile(self, tile_id):
+        """ Reprocess a single failed tile
+
+        :param tile_id: An ID of a tile
+        :type tile_id: int or None
+        """
+        self._call_job(f'tiles/{tile_id}/restart')
```

Comment:
This might be the reason why job names are not enums, but I'd argue that this method could still be calling job `f'{TILES}/{tile_id}/{RESTART}'` or similar.

Reply:
Similarly to above - I don't think it is used often enough to make it an enum or a constant.